src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -234,6 +234,70 @@ def calculate(
         )
 
 
+class TemperatureCalculator(MetricCalculator[Sample[Temperature]]):
+    """Define how to calculate temperature metrics."""
+
+    def __init__(self, batteries: Set[int]) -> None:
+        """Create class instance.
+
+        Args:
+            batteries: What batteries should be used for calculation.
+        """
+        super().__init__(batteries)
+
+        self._metrics = [
+            ComponentMetricId.TEMPERATURE_MAX,
+        ]
+
+    @classmethod
+    def name(cls) -> str:
+        """Return name of the calculator.
+
+        Returns:
+            Name of the calculator
+        """
+        return "temperature"
+
+    @property
+    def battery_metrics(self) -> Mapping[int, list[ComponentMetricId]]:
+        """Return what metrics are needed for each battery.
+
+        Returns:
+            Map between battery id and set of required metrics id.
+        """
+        return {bid: self._metrics for bid in self._batteries}
+
+    @property
+    def inverter_metrics(self) -> Mapping[int, list[ComponentMetricId]]:
+        """Return what metrics are needed for each inverter.
+
+        Returns:
+            Map between inverter id and set of required metrics id.
+        """
+        return {}
+
+    def calculate(
+        self,
+        metrics_data: dict[int, ComponentMetricsData],
+        working_batteries: set[int],
+    ) -> Sample[Temperature] | None:
+        timestamp = _MIN_TIMESTAMP
+        temperature_max: float = 0
+        for battery_id in working_batteries:
+            if battery_id not in metrics_data:
+                continue
+            metrics = metrics_data[battery_id]
+            temperature = metrics.get(ComponentMetricId.TEMPERATURE_MAX)
+            if temperature is None:
+                continue
+            timestamp = max(timestamp, metrics.timestamp)
+            temperature_max = max(temperature_max, temperature)
+        if timestamp == _MIN_TIMESTAMP:
+            return None
+
+        return Sample(timestamp=timestamp, value=Temperature(value=temperature_max))
```

Comment:
This constructor shouldn't work, as the default constructor is disabled.

```suggestion
        return Sample(timestamp=timestamp, value=Temperature.from_celsius(temperature_max))
```